lib/_http_outgoing.js

```diff
@@ -602,6 +603,24 @@ OutgoingMessage.prototype.getHeaderNames = function getHeaderNames() {
 };
 
 
+// Returns an array of the names of the current outgoing raw headers.
+OutgoingMessage.prototype.getRawHeaderNames = function getRawHeaderNames() {
+  const headersMap = this[kOutHeaders];
+  const headers = [];
+
+  if (headersMap !== null) {
+    const values = ObjectValues(headersMap);
+    // Retain for(;;) loop for performance reasons
+    // Refs: https://github.com/nodejs/node/pull/30958
+    for (let i = 0, l = values.length; i < l; i++) {
+      headers.push(values[i][0]);
```

Comment:
I think we should use `%Array.prototype.map%` (my guess is it performs just as good, but we would need to run benchmark to know), but if you prefer to keep the `for(;;)` loop, you need to make sure you code is not using any user-mutable method.

```suggestion
      ArrayPrototypePush(headers, values[i][0]);
```
 or

```suggestion
      headers[i] = values[i][0];
```

Reply:
Got it that makes sense :+1: